File: scripts/headless_console_check.py
import sys
from playwright.sync_api import sync_playwright

def main():
    base_url = sys.argv[1] if len(sys.argv) > 1 else 'http://localhost:5173'
    routes = ['/', '/task-board', '/video-match', '/video-analysis']
    
    print(f"Starting headless browser check at base URL: {base_url}")
    
    warnings = []
    errors = []
    
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        
        for route in routes:
            url = f"{base_url}{route}"
            print(f"  Checking {url} ...")
            page = browser.new_page()
            
            def handle_console(msg):
                text = msg.text
                if '[vite]' in text:
                    return
                if msg.type in ('warning', 'warn'):
                    if '[Vue warn]' in text or 'warning' in text.lower():
                        warnings.append(f"[{route}] {text}")
                elif msg.type == 'error':
                    errors.append(f"[{route}] {text}")
                    
            def handle_pageerror(err):
                errors.append(f"[{route}] {err.message}")
                
            page.on("console", handle_console)
            page.on("pageerror", handle_pageerror)
            
            page.add_init_script("""
                window.addEventListener('unhandledrejection', event => {
                    const msg = event.reason && event.reason.message ? event.reason.message : String(event.reason);
                    console.error('Unhandled Promise Rejection: ' + msg);
                });
            """)
            
            try:
                page.goto(url, wait_until="networkidle", timeout=15000)
                page.wait_for_timeout(2000)
            except Exception as e:
                print(f"    Failed to navigate to {url}: {e}")
                errors.append(f"[{route}] Navigation Failed: {e}")
                
            page.close()
            
        browser.close()
        
    if errors or warnings:
        print('\n--- HEADLESS CONSOLE CHECK FAILED ---')
        if errors:
            print('\nErrors found:')
            for e in errors:
                print(f"  ❌ {e}")
        if warnings:
            print('\nWarnings found:')
            for w in warnings:
                print(f"  ⚠️ {w}")
        sys.exit(1)
        
    try:
        print('✅ Headless console check passed! No Vue warnings or errors detected.')
    except UnicodeEncodeError:
        print('OK! Headless console check passed! No Vue warnings or errors detected.')
    sys.exit(0)

if __name__ == "__main__":
    main()

File: scripts/headless_console_check.js
import puppeteer from 'puppeteer';

(async () => {
  const url = process.argv[2] || 'http://localhost:5173';
  console.log(`Starting headless browser check at ${url}...`);

  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });

  const page = await browser.newPage();
  
  const warnings = [];
  const errors = [];

  // Listen to console events
  page.on('console', msg => {
    const type = msg.type();
    const text = msg.text();
    
    // Ignore vite dev server logs
    if (text.includes('[vite]')) return;
    
    if (type === 'warning' || type === 'warn') {
      if (text.includes('[Vue warn]') || text.includes('warning')) {
        warnings.push(text);
      }
    } else if (type === 'error') {
      // Ignore some non-critical errors if any, but capture Vue errors and generic TypeErrors
      errors.push(text);
    }
  });

  page.on('pageerror', err => {
    errors.push(err.message);
  });

  try {
    // Navigate and wait for network to be idle
    await page.goto(url, { waitUntil: 'networkidle0', timeout: 15000 });
    
    // Wait a brief moment for any final async renderings or transitions
    await new Promise(resolve => setTimeout(resolve, 2000));
    
  } catch (err) {
    console.error(`Failed to navigate to ${url}: ${err.message}`);
    await browser.close();
    process.exit(1);
  }

  await browser.close();

  if (errors.length > 0 || warnings.length > 0) {
    console.log('\n--- HEADLESS CONSOLE CHECK FAILED ---');
    if (errors.length > 0) {
      console.log('\nErrors found:');
      errors.forEach(e => console.log(`  ❌ ${e}`));
    }
    if (warnings.length > 0) {
      console.log('\nWarnings found:');
      warnings.forEach(w => console.log(`  ⚠️ ${w}`));
    }
    process.exit(1);
  }

  console.log('✅ Headless console check passed! No Vue warnings or errors detected.');
  process.exit(0);
})();
